bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\Date\Value as DateValue;
+use Ibexa\Core\FieldType\DateAndTime\Value as DateAndTimeValue;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Configuration\ScheduledVisibilityConfiguration;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\VisibilityUpdateResult;
+use OutOfBoundsException;
+
+use function in_array;
+
+final class ScheduledVisibilityService
+{
+    public function __construct(
+        private readonly ScheduledVisibilityConfiguration $configurationService,
+    ) {}
+
+    /**
+     * @throws OutOfBoundsException
+     */
+    public function updateVisibilityIfNeeded(Content $content): VisibilityUpdateResult
```

Comment:
Please add here a second argument `string $handlerIdentifier = null` and fetch the handler from the registry. Configuration doesn't need to return the handler directly.

Reply:
fixed here 592fadbd0cddf1fca44b7bf5b234446368ab5bfb

Reply:
My idea was to use the optional parameter to override the configured handler if needed, basically:

```php
$handler = $this->registry->get($handlerIdentifier ?? $this->configurationService->getHandlerIdentifier());
```

Reply:
fixed in fa8e30b8c58ae4f1dba7b5e1d5facef7e900687c